libs/datasets/new_cases_and_deaths.py

```diff
@@ -72,36 +72,39 @@ def add_new_deaths(dataset_in: MultiRegionDataset) -> MultiRegionDataset:
 def spread_first_reported_value_after_stall(
     series: pd.Series, max_days_to_spread: int = 7
 ) -> pd.Series:
-    """Spreads first reported value after reported zeros.
+    """Spreads first reported value after reported zeros by dividing it evenly
+    over the prior days.
 
     Args:
         series: Series of new cases with date index.
-        max_days_to_spread: Maximum number of days to spread a single report.
+        max_days_to_spread: Maximum number of days to spread a single report. If
+            a stall exceeds this number of days, the reported value will be spread
+            over max_days_to_spread and the remaining zeros will be kept as
+            zeros.
     """
-    # Find points in the series that are either zeros or the first report after a string of zeros.
     if series.first_valid_index() is None:
         return series
 
     # Counting consecutive zeros
     zeros = series == 0
     zeros_count = zeros.cumsum()
 
-    stalled_cases_count = zeros_count.sub(zeros_count.mask(zeros).ffill().fillna(0))
+    stalled_days_count = zeros_count.sub(zeros_count.mask(zeros).ffill().fillna(0))
 
     # Add one more day for spreading on first report after zeros
     first_report_after_zeros = (series != 0) & (series.shift(1) == 0)
-    stalled_cases_count = stalled_cases_count + (
-        (stalled_cases_count.shift(1) + 1) * first_report_after_zeros
+    stalled_days_count = stalled_days_count + (
+        (stalled_days_count.shift(1).fillna(0) + 1) * first_report_after_zeros
```

Comment:
Not sure if the `fillna(0)` is actually necessary but it seemed to make sense to keep `NaN`s out...